modules/storage-api/src/main/java/org/apache/ignite/internal/storage/index/IndexBinaryRow.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.storage.index;
+
+import org.apache.ignite.internal.schema.BinaryRow;
+import org.apache.ignite.internal.storage.PartitionStorage;
+
+/**
+ * Represents an Index Row - a set of indexed columns and Primary Key columns (for key uniqueness).
```

Comment:
It was true for Ignite-2 that PK columns were attached for the uniqueness purposed.
Now, PK columns are attached for searching in the data tree purposes, isn't is?

Reply:
Not. It isn't.
The uniqueness and lookup full entry by the index record are independent.